### PR TITLE
IPFS Gateway API: /ipfs/hash endpoint should work per file

### PR DESCRIPTION
**Pull request by mbrizic**

### CODE REVIEW

File: nodes/persistence-node/src/utils/isTextFile.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const textExtensions = [".graphql", ".yaml"]
```
Comment (mbrizic) [inline nodes/persistence-node/src/utils/isTextFile.ts:1]: @nerfZael Should we cover other extensions here, or is it expected that wrappers will only contain those two?

**Comment by nerfZael** [comment]
I've should've been clearer with this issue, we want to mimick: https://ipfs.io/ipfs/QmPSuejkLTkcbs8RMJbwxAQBS2FqQAD1eqsLdC19o2Lv1Q
Which means we should support a "path" after "/ipfs/"
In the above example it could be: "/ipfs/QmPSuejkLTkcbs8RMJbwxAQBS2FqQAD1eqsLdC19o2Lv1Q/meta/imgs/github.svg"
It should show a directory view (like the one we already have) if it's a directory, the text/image file if it's text/image and download it if it is binary. I think we can let the browser take care of file display/download (instead of us checking the extension) and just check if it's a file or directory

**Comment by nerfZael** [comment]
> I've should've been clearer with this issue, we want to mimick: https://ipfs.io/ipfs/QmPSuejkLTkcbs8RMJbwxAQBS2FqQAD1eqsLdC19o2Lv1Q Which means we should support a "path" after "/ipfs/" In the above example it could be: "/ipfs/QmPSuejkLTkcbs8RMJbwxAQBS2FqQAD1eqsLdC19o2Lv1Q/meta/imgs/github.svg" It should show a directory view (like the one we already have) if it's a directory, the text/image file if it's text/image and download it if it is binary. I think we can let the browser take care of file display/download (instead of us checking the extension) and just check if it's a file or directory

You should be able to use something like ` await ipfsNode.files.stat("path");` to get info about a file or directory, it has a type property that says "file" or "directory"
Note you need to have the path start with "/ipfs"

**Comment by nerfZael** [comment]
Uses old code to get the contents of the file, but should work
```
app.get('/ipfs/*', handleError(async (req, res) => {
      
      const ipfsPath = req.url;

      const a = await ipfs.files.stat(`${ipfsPath}`);
      if(a.type === "file") {
        const stream = ipfs.cat(ipfsPath);

        let data: Uint8Array = new Uint8Array();
  
        for await (const chunk of stream) {
          const temp = new Uint8Array(data.length + chunk.length);
          temp.set(data);
          temp.set(chunk, data.length);
          data = temp;
        }
  
        const buffer = Buffer.from(data);

        res.end(buffer);
        return;
      } else if(a.type === "directory") {

        const files = await asyncIterableToArray(
          ipfs.ls(ipfsPath)
        );
  
        res.render('ipfs-directory-contents', {
          files,
          ipfsPath,
          sizeInKb: function () {
            return formatFileSize((this as any).size)
          }
        });
      } else {
        throw new Error(`Unknown file type: ${a.type}`);
      }
    }));
```

File: nodes/persistence-node/src/services/IpfsGatewayApi.ts [hunk 2]
```diff
@@ -114,20 +104,30 @@ export class IpfsGatewayApi {
       })
     }));
 
-    app.get('/ipfs/:hash', handleError(async (req, res) => {
+    app.get("/ipfs/:hash(*)", handleError(async (req, res) => {
       const hash = (req.params as any).hash as string;
```
Comment (nerfZael) [inline nodes/persistence-node/src/services/IpfsGatewayApi.ts:108]: its an ipfsPath, not a hash, it can be e.g. ""Qm......sd/meta/schema.graphql"

**Review by nerfZael — CHANGES_REQUESTED** [review]

File: nodes/persistence-node/src/services/IpfsGatewayApi.ts (hunk 2 above)
Comment (mbrizic) [reply nodes/persistence-node/src/services/IpfsGatewayApi.ts]: Fixed.

**Review by nerfZael — APPROVED** [review]

**Pull request merged**